### DISCUSSION TITLE
Timing of "deploy" with new option in mirthSync 3.0.1

### DISCUSSION THREAD

**Discussion by elavy-harris** [nextgenhealthcare/connect#5179] [Q&A] [answered]
Not sure if this is the right forum, but https://forums.mirthproject.io/forum/open-source-hit/general-discussion-aa/17664-mirthsync-git-source-control-and-config-management points here.

The help text for the new "--deploy" flag for mirthSync says, "deploy each included channel immediately after saving the channel to Mirth."

That sounds like a problem, as Code Templates are pushed after Channels, so the deploy might occur before the new Code Template code has been pushed.  Is the description wrong (i.e., the deploy happens after the rest of the push is complete) or does the deploy really happen immediately after saving each channel?

**Comment by jonbartels** [comment, marked as answer]
https://github.com/SagaHealthcareIT/mirthsync would be a better venue for the question. 

The relevant commit seems to be https://github.com/SagaHealthcareIT/mirthsync/commit/5d9aa5534c105d0835e91f101650d79ac6ad5b61
